io/src/main/scala/sbt/io/IO.scala

```diff
@@ -454,13 +462,35 @@ object IO {
 
   /** Deletes `file`, recursively if it is a directory. */
   def delete(file: File): Unit = {
-    translate("Error deleting file " + file + ": ") {
-      val deleted = file.delete()
-      if (!deleted && file.isDirectory) {
-        delete(listFiles(file))
-        file.delete
-        ()
+    object deleter extends SimpleFileVisitor[NioPath] {
+      override def visitFile(file: NioPath, attr: BasicFileAttributes): FileVisitResult = {
+        translate("Error deleting file " + file.toFile + ": ") {
```

Comment:
as translate is a exception wrapper would it be better to have just one wrapper?

Reply:
It's up to you how to organize the code; I'm just pointing out the reason for the failures described at https://github.com/sbt/io/pull/133#pullrequestreview-99974172.